rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -0,0 +1,601 @@
+        cs.addSrcIdx(unop->srcIdx);                                            \
+        store(it, unop);                                                       \
+        break;                                                                 \
+    }
+                UNOP(Plus, uplus);
+                UNOP(Minus, uminus);
+                UNOP(Not, Not);
+                UNOP(Length, length);
+#undef UNOP
+
+            case Tag::Is: {
+                assert(false &&
+                       "is_ takes an immediate, but in PIR it is unop");
+                assert(false && "not yet implemented.");
+                break;
+            }
+            case Tag::Call: {
+                auto call = Call::Cast(instr);
+                auto cls = call->cls();
+                loadEnv(it, call->env());
+                cs << BC::setEnv()
+                   << BC::ldloc(alloc[cls]); // TODO: can use load(it, cls) for
+                                             // the closure?
+
+                std::vector<FunIdxT> callArgs;
+                call->eachCallArg([&](Value* arg) {
+                    auto mkarg = MkArg::Cast(arg);
+                    callArgs.push_back(promises[mkarg->prom]);
+                });
+
+                cs.insertCall(Opcode::call_, callArgs, {},
+                              Pool::get(call->srcIdx));
+                store(it, call);
+                break;
+            }
+            case Tag::StaticCall: {
+                assert(false && "not yet implemented.");
+                break;
+            }
+            case Tag::StaticEagerCall: {
+                assert(false && "not yet implemented.");
+                break;
+            }
+            case Tag::CallBuiltin: {
+                auto blt = CallBuiltin::Cast(instr);
+                // TODO: is it needed to set the environment?
+                loadEnv(it, blt->env());
+                cs << BC::setEnv();
+                blt->eachCallArg([&](Value* arg) { load(it, arg); });
+                cs.insertStackCall(Opcode::static_call_stack_, blt->nCallArgs(),
+                                   {}, Pool::get(blt->srcIdx), blt->blt);
+                store(it, blt);
+                break;
+            }
+            case Tag::CallSafeBuiltin: {
+                assert(false && "not yet implemented.");
+                break;
+            }
+            case Tag::MkEnv: {
+                auto mkenv = MkEnv::Cast(instr);
+                loadEnv(it, mkenv->parent());
+                cs << BC::makeEnv() << BC::dup() << BC::setEnv();
+                // bind all args
+                mkenv->eachLocalVar([&](SEXP name, Value* val) {
+                    load(it, val);
+                    cs << BC::stvar(name);
+                });
+                store(it, mkenv);
+                break;
+            }
+            case Tag::Phi: {
+                auto phi = Phi::Cast(instr);
+                load(it, phi->arg(0).val());
```

Comment:
I don't understand this. Why can you take the input 0? because you know that all of the inputs are in the same local value?